src/engine/Map.hxx

```diff
@@ -87,13 +87,21 @@ public:
         break;
       }
     }
+
     if (isOkToSet)
     {
       int groundtileIndex = -1;
       for (auto it = begin; it != end; ++it)
       {
         bool shouldRender = !(!isMultiObjects && it != begin);
         Layer layer = TileManager::instance().getTileLayer(tileID);
+
+        // During road construction, do not place new road tile over the old one except it is the only tile for construction
+        if ((layer == ROAD) && mapNodes[it->x * m_columns + it->y]->isLayerOccupied(layer) && ((begin + 1) != end))
+        {
+          continue;
+        }
+
```

Comment:
this should be done inside MapNode::setTileID instead of here, more organized.
```
switch (layer)
    {
    case Layer::WATER:
      demolishLayer(Layer::ROAD);
      // no break on purpose. 

@Bogi79 **add a break here also demolishLayer(Layer::BUILDINGS);**

    case Layer::ROAD:
      // in case it's allowed then maybe a Tree Tile already exist, so we remove it.

@Bogi79 **should be done here**, in case Layer::ROAD occupied return from function.

      demolishLayer(Layer::BUILDINGS);
      break;
```

Reply:
I would agree that place, where I put the change, is not the best one. But pushing the logic even deeper in the class hierarchy seems even worse. In my opinion, a few extra layers are missing in between, where such filtering would be more appropriate.

Additionally, on suggested place, the road tile could not be overwritten with another one without destroying the existing one previously.